File: backend/src/main/kotlin/com/trackwatch/backend/service/EmailService.kt
package com.trackwatch.backend.service

import com.trackwatch.backend.clients.email.ResendClient
import com.trackwatch.backend.model.Track
import com.trackwatch.backend.model.User
import com.trackwatch.backend.utils.service.EmailHelper
import org.springframework.stereotype.Service

@Service
class EmailService(
    private val resendClient: ResendClient,
    private val emailHelper: EmailHelper
) {

    fun sendWelcomeEmail(user: User) {
        val subject = emailHelper.generateWelcomeEmailSubject()
        val htmlBody = emailHelper.generateWelcomeEmailBody(user)
        resendClient.sendEmail(user, subject, htmlBody)
    }

    fun sendAddedTracksEmail(user: User, addedTracks: List<Track>) {
        if (addedTracks.isNotEmpty()) {
            val subject = emailHelper.generateAddedTracksEmailSubject(addedTracks)
            val htmlBody = emailHelper.generateAddedTracksEmailBody(addedTracks)
            resendClient.sendEmail(user, subject, htmlBody)
        }
    }

}

File: backend/app/services/email_service.py
from app.utils.email_helper import *
from app.clients.email.resend_client import send_email, send_admin_email
from decouple import config

def send_welcome_email(user):
  subject = generate_welcome_email_subject()
  html_body = generate_welcome_email_body(user)
  return send_email(user, subject, html_body)

def send_added_tracks_email(user, added_tracks):
  if added_tracks:
    subject = generate_added_tracks_email_subject(added_tracks)
    html_body = generate_added_tracks_email_body(added_tracks)
    return send_email(user, subject, html_body)
  return False

def send_spotify_reauth_reminder_email(user, expires_at, days_remaining):
  subject = generate_spotify_reauth_reminder_email_subject(days_remaining)
  html_body = generate_spotify_reauth_reminder_email_body(user, expires_at, days_remaining)
  return send_email(user, subject, html_body)

def send_admin_notification_email(users_count, errors=None):
  admin_email = config('ADMIN_EMAIL', default=None)

  if not admin_email:
    print("ADMIN_EMAIL environment variable not set")
    return

  subject = generate_admin_notification_email_subject()
  html_body = generate_admin_notification_email_body(users_count, errors)

  return send_admin_email(admin_email, subject, html_body)
